litestar/app.py

```diff
@@ -466,18 +466,24 @@ def serialization_plugins(self) -> list[SerializationPluginProtocol]:
         return list(self.plugins.serialization)
 
     @staticmethod
-    def _get_default_plugins() -> list[PluginProtocol]:
-        default_plugins: list[PluginProtocol] = []
+    def _get_default_plugins(plugins: list[PluginProtocol] | None = None) -> list[PluginProtocol]:
+        if plugins is None:
+            plugins = []
         with suppress(MissingDependencyException):
-            from litestar.contrib.pydantic import PydanticInitPlugin, PydanticSchemaPlugin
-
-            default_plugins.extend((PydanticInitPlugin(), PydanticSchemaPlugin()))
+            from litestar.contrib.pydantic import PydanticInitPlugin, PydanticPlugin, PydanticSchemaPlugin
 
+            pre_configured = any(
+                isinstance(plugin, (PydanticPlugin, PydanticInitPlugin, PydanticSchemaPlugin)) for plugin in plugins
```

Comment:
If the user provides a `PydanticInitPlugin` but not a `PydanticSchemaPlugin` (or vice-versa) then it won't get added right? Should this be handled?

Reply:
My thought process was that if the user has supplied any plugin, then they likely want to override the default behavior.  Maybe that's not a good way to think about it though?

Reply:
If they give `PydanticPlugin` then yeah I agree with you. As a user I wouldn't expect passing in a custom `PydanticInitPlugin` to disable the `PydanticSchemaPlugin`.

Reply:
Let me see whatI an do to address this.

Reply:
@guacs I think this should handle this scenario properly now.